service-config/src/main/java/com/palantir/remoting2/config/service/ProxyConfiguration.java

```diff
@@ -35,30 +35,55 @@
 @JsonSerialize(as = ImmutableProxyConfiguration.class)
 @Style(visibility = Style.ImplementationVisibility.PACKAGE, builder = "new")
 public abstract class ProxyConfiguration {
+    enum Type {
+        direct, http;
+    }
 
     /**
      * The hostname and port of the HTTP/HTTPS Proxy. Recognized formats include those recognized by {@link
      * com.google.common.net.HostAndPort}, for instance {@code foo.com:80}, {@code 192.168.3.100:8080}, etc.
      */
-    public abstract String hostAndPort();
+    public abstract Optional<String> hostAndPort();
```

Comment:
true. I can add a (deprecated) method that returns a non-optional String and throws when the type is direct. That should not break any existing consumers.